subgraph/src/datasources/FortaStaking.ts

```diff
@@ -0,0 +1,248 @@
+import { Address, BigInt, ethereum } from "@graphprotocol/graph-ts";
+import { logStore, newMockEvent } from "matchstick-as";
+import {
+  StakeDeposited as StakeDepositedEvent,
+  Rewarded as RewardedEvent,
+  Slashed as SlashedEvent,
+  Froze as FrozeEvent,
+  FortaStaking as FortaStakingContract,
+} from "../../generated/FortaStaking/FortaStaking";
+import { Reward, Slash, Stake, Staker, Subject } from "../../generated/schema";
+import { fetchAccount } from "../fetch/account";
+
+import eventId from "../utils/event";
+
+export function handleStakeDeposited(event: StakeDepositedEvent): void {
+  let subject = Subject.load(event.params.subject.toHex());
+
+  let staker = Staker.load(event.params.account.toHex());
+  let stake = Stake.load(eventId(event));
+  const fortaStaking = FortaStakingContract.bind(event.address);
+
+  if (!subject) {
+    subject = new Subject(event.params.subject.toHex());
+    subject.activeStake = fortaStaking.activeStakeFor(
+      event.params.subjectType,
+      event.params.subject
+    );
+    subject.inactiveStake = fortaStaking.inactiveStakeFor(
+      event.params.subjectType,
+      event.params.subject
+    );
+
+    subject.inactiveShares = fortaStaking.inactiveSharesOf(
+      event.params.subjectType,
+      event.params.subject,
+      event.params.account
+    );
+
+    subject.activeShares = fortaStaking.sharesOf(
+      event.params.subjectType,
+      event.params.subject,
+      event.params.account
+    );
+
+    subject.isFrozen = false;
+    subject.slashedTotal = 0;
+    subject.subjectType = event.params.subjectType;
+    subject.save();
+  }
+
+  if (!staker) {
+    staker = new Staker(event.params.account.toHex());
+    staker.save();
+  }
+  if (!stake) {
+    stake = new Stake(eventId(event));
+    stake.subjectId = event.params.subject.toHex();
+    stake.subjectType = event.params.subjectType;
+    stake.isActive = true;
+    stake.staker = event.params.account.toHex();
+
+    stake.stake = event.params.amount;
+    stake.shares = fortaStaking.sharesOf(
+      event.params.subjectType,
+      event.params.subject,
+      event.params.account
+    );
+    stake.save();
+  }
+}
+
+export function handleRewarded(event: RewardedEvent): void {
+  let reward = new Reward(eventId(event));
```

Comment:
Rewards are an @entity that does NOT inherit from @event. Is that really what you want ? If so, why use the event naming convention, its confusing.

Reply:
Rewards are indeed entities, I'm tracking the Rewarded event that is in the contract abi, so that I can successfully create a Reward entity